src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -130,28 +173,76 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string paramete
                     this.logger?.LogDebug("Running PowerShell rule engine");
                     evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
                 }
+               
+                // Code to handle nested templates recursively
+                dynamic jsonTemplate = JsonConvert.DeserializeObject(template);
```

Comment:
Not necessarily. I don't really deeply understand the underlying differences between the typing so I used whichever ones worked when I did not know the explicit types. Any suggestions for what to use instead?

Reply:
Documentation about `dynamic`: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/using-type-dynamic
We tend to use `var`: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/var

Reply:
Ok, I changed all the dynamics to vars where the change did not break anything. I only have dynamic in a few places where if I switched some fields would not be accessible.

Reply:
Instead of using `JsonConvert.DeserializeObject`, use `JObject.Parse` which will return a JToken, resolving the need for `dynamic`.  `DeserializeObject` returns a dynamic object, so the object properties are not known at compilation.  `JToken`s have known properties and use string indexing to traverse JSON properties, which is safer.

Reply:
Actually, since `JObject.Parse` is already done in `templateContext.OriginalTemplate`, just use that and remove `jsonTemplate` altogether.

Reply:
The template parsed in templateContext is different, so I kept jsonTemplate but replaced DeserializeObject with JObject.Parse